src/v/cluster/topic_updates_dispatcher.cc

```diff
@@ -53,32 +53,33 @@ topic_updates_dispatcher::apply_update(model::record_batch b) {
                       del_cmd.key, *topic_assignments);
                 }
                 return dispatch_updates_to_cores(del_cmd, base_offset)
-                  .then([this,
-                         tp_ns = std::move(del_cmd.key),
-                         topic_assignments = std::move(topic_assignments),
-                         in_progress = std::move(in_progress),
-                         allocation_domain = get_allocation_domain(
-                           del_cmd.key)](std::error_code ec) {
-                      if (ec == errc::success) {
-                          vassert(
-                            topic_assignments.has_value(),
-                            "Topic had to exist before successful delete");
-                          deallocate_topic(
-                            *topic_assignments, in_progress, allocation_domain);
+                  .then(
+                    [this,
+                     tp_ns = std::move(del_cmd.key),
+                     topic_assignments = std::move(topic_assignments),
+                     in_progress = std::move(in_progress)](std::error_code ec) {
+                        if (ec == errc::success) {
```

Comment:
nice catch.

prior to this change get_allocation_domain ran before the continuation (when the capture list was created). but now it runs when the continuation is scheduled. is that something to be concerned with? in this case it might make sense to remove `std::move` from `tp_ns` in the capture list and pay a small string copy cost to keep things simple and preserve the semantics.

Reply:
No concern here. `tp_ns` is not changed (the capture is not mutable), the lambda in not a coro, and `get_allocation_domain()` is a pure function.

Reply:
> No concern here. tp_ns is not changed (the capture is not mutable), the lambda is not a coro, and get_allocation_domain() is a pure function.

awesome. thanks for the explanation